powerfulseal/k8s/k8s_inventory.py

```diff
@@ -47,6 +47,19 @@ def is_fresh(self, when):
             return False
         return True
 
+    def get_regex_namespaces(self, regex):
+        self.logger.info("regex namespace top")
+        regex_no_star = regex.replace("*", '')
+        namespaces = []
+        for item in self.k8s_client.list_namespaces():
+            expr = re.compile(regex, re.IGNORECASE)
+            if expr:
+                regex_any_match = expr.search(str(item.metadata.name))
+                # Make sure search is not not and see if full match is equal to regex string
+                if regex_any_match and regex_any_match.group(0) == regex_no_star:
+                    namespaces.append(item.metadata.name)
```

Comment:
Thanks for refactoring, it does look cleaner.

I would prefer to skip the `regex_any_match.group(0) == regex_no_star` check, to make it more flexible.

Let's say that you have a namespace "something-app-somethingelse".

By default, using value `.*app.*` will match all namespaces containing `app` in them. Using something like `^something-.*-somethingelse$` I can be more specific.

Also, if I query for `something-.*`, I would like it to work with `something-else` as well as `something-12345`, not just `something-.X` - I haven't tested it, but from reading it, I think it doesn't?

Finally, it would be really great, if you could check in a few tests for this method. It would illustrate how it works.

Reply:
I added a test file and was able to get your example to work properly. I did have to take 

I tried to find any matches with "openshift-t*" against namespaces: "openshift-test", "openshift-two", "openshift-three", "openshift-four". Without that check it matches on "openshift-four" when it's not supposed to. But adding the ".*" worked on matching the proper strings.